substrate/frame/support/procedural/src/pallet/parse/config.rs

```diff
@@ -406,10 +440,20 @@ impl ConfigDef {
 			false
 		};
 
-		let has_frame_system_supertrait = item.supertraits.iter().any(|s| {
-			syn::parse2::<syn::Path>(s.to_token_stream())
-				.map_or(false, |b| has_expected_system_config(b, frame_system))
-		});
+		let (has_frame_system_supertrait, has_event_bound) = item
+			.supertraits
+			.iter()
```

Comment:
We should do what @gui1117 proposed and introduce a `frame_system_pallet` attribute macro that we will use in `frame_system`. Every other pallet will then be assumed to not be `frame-system`.